tests/test_executor.py

```diff
@@ -3,86 +3,109 @@
 from typing import Optional
 from pathlib import Path
 from tempfile import TemporaryDirectory
+from typing import Optional
+
+from psij import SubmitException, Job, JobExecutor, JobSpec, JobState, JobStatus, JobExecutorConfig
+from psij.executors.slurm import SlurmExecutorConfig
+from executor_test_params import ExecutorTestParams
+
+
+def assert_completed(job: Job) -> None:
+    if job.status.state == JobState.FAILED:
+        raise RuntimeError('Job (native_id: {}) failed: {}'.format(job.native_id,
+                                                                   job.status.message))
+    elif job.status.state == JobState.COMPLETED:
+        return
+    else:
+        raise RuntimeError('Unexpected job state: {}'.format(job.status.state))
 
 from psij import SubmitException, Job, JobExecutor, JobSpec, JobState
 
-tests = [['local', None],
-         ['saga', 'fork://localhost/'],
-         ['saga', 'ssh://localhost/']]
+
+def _get_executor_instance(ep: ExecutorTestParams, job: Job) -> JobExecutor:
+    assert job.spec is not None
+    job.spec.launcher = ep.launcher
+    return JobExecutor.get_instance(ep.executor, url=ep.url)
 
 
-@pytest.mark.parametrize('name,url', tests)
-def test_simple_job(name: str, url: Optional[str]) -> None:
-    job = Job(JobSpec(executable='/bin/date'))
-    jex = JobExecutor.get_instance(name=name, url=url)
-    jex.submit(job)
+def test_simple_job(execparams: ExecutorTestParams) -> None:
+    job = Job(JobSpec(executable='/bin/date', launcher=execparams.launcher))
+    exec = _get_executor_instance(execparams, job)
+    exec.submit(job)
     job.wait()
 
 
-@pytest.mark.parametrize('name,url', tests)
-def test_simple_job_redirect(name: str, url: Optional[str]) -> None:
-    with TemporaryDirectory() as td:
+def test_simple_job_redirect(execparams: ExecutorTestParams) -> None:
+    with TemporaryDirectory(dir=Path.home() / '.psij' / 'work') as td:
         outp = Path(td, 'stdout.txt')
-        job = Job(JobSpec(executable='/bin/echo', arguments=['-n', '_x_'],
-                          stdout_path=outp))
-        jex = JobExecutor.get_instance(name=name, url=url)
-        jex.submit(job)
+        job = Job(JobSpec(executable='/bin/echo', arguments=['-n', '_x_'], stdout_path=outp))
+        exec = _get_executor_instance(execparams, job)
+        exec.submit(job)
         job.wait()
         f = outp.open("r")
         contents = f.read()
         assert contents == '_x_'
 
 
-@pytest.mark.parametrize('name,url', tests)
-def test_attach(name: str, url: Optional[str]) -> None:
+def test_attach(execparams: ExecutorTestParams) -> None:
     job = Job(JobSpec(executable='/bin/sleep', arguments=['1']))
-    jex = JobExecutor.get_instance(name=name, url=url)
-    jex.submit(job)
-    job.wait(target_states=[JobState.ACTIVE])
+    exec = _get_executor_instance(execparams, job)
+    exec.submit(job)
```

Comment:
`exec`  is a keyword (applies elsewhere too)

Reply:
Thank you. Will update.